modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -243,6 +245,7 @@ public TableColumnHeader(final TableColumnBase tc) {
         }
 
         if (me.isConsumed()) return;
+        popupTriggeredOnMousePressed = me.isPopupTrigger();
         me.consume();
 
         header.getTableHeaderRow().columnDragLock = true;
```

Comment:
On Mac a context menu can be invoked using the right button or the left button in conjunction with the Control key. In the second case the pop trigger flag will be set and the primary button will be down. If I'm reading the code correctly this could get the header stuck in column re-ordering mode.

Reply:
control+left-click works as expected for me too.

When the context menu is showing, clicking anywhere outside of it, closes the popup (because of the autohide property), but doesn't consume the event. If you click on a button, for instance, it will be fired. In any case, this is the same before and after this PR, so maybe an issue for a later discussion.

Reply:
> control+left-click works as expected for me too.

For control+left-click it looks like the code will call `columnReordingStarted` but on the released event it won't call `columnReorderingComplete`. I have no idea if this will lead to problems or not.

Reply:
@beldenfox Do you think we should address that issue (ctrl+left click + columnReordering) in this PR? Else, do we need to do anything else to move it forward?

Reply:
You should be good with a few tweaks. If the press event is a popup trigger don't set the columnDragLock or call columnReorderingStarted (since you might not get a release event). In the drag handler check popupTriggeredOnMousePressed and do nothing if it's set. Your released handler looks fine as-is. I think that should cover it.

By the way, in a perfect world you would have more predictable behavior. When the popup menu is shown it would grab the mouse pointer and you would never see the release event. That would allow the user to click to bring up the popup and start dragging across the menu items without having to release the mouse.

Reply:
Thanks, @beldenfox! I've included your suggestions,  and also added the ctrl+left click macOS case to the test.